### PR TITLE
feat(ZMSKVR-707 ZMSKVR-508): list view date accordion add the earlier and later buttons like in calendar view. also cleanup some styling.

### PR DESCRIPTION
**Pull request by ThomasAFink**
### Pull Request Checklist (Feature Branch to `next`):

- [x] Ich habe die neuesten Änderungen aus dem `next` Branch in meinen Feature-Branch gemergt.
- [x] Das Code-Review wurde abgeschlossen.
- [x] Fachliche Tests wurden durchgeführt und sind abgeschlossen.

## Summary by CodeRabbit

- **New Features**
  - Enhanced appointment list view: per-day navigation (earlier/later for hours and AM/PM), per-day state preservation, automatic opening of first available day, and improved accordion accessibility/aria behavior.

- **Style**
  - Added global responsive breakpoint utilities and visibility helper classes for consistent behavior across screen sizes; updated styles for list-view controls.

- **Tests**
  - Extensive unit tests for list-view navigation, provider filtering, calendar toggles, snapping/edge cases, and multi-provider rendering.

- **Chores**
  - Added Sass dev dependency, automatic SCSS import of breakpoints via build config, and adjusted TypeScript module resolution settings.

### CODE REVIEW

File: zmscitizenview/src/styles/breakpoints.scss [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  @include xs-down {
+    display: none !important;
+  }
+}
+
+.hidden-sm-up {
+  @include xs-up {
+    display: none !important;
+  }
+}
```
[the remark this answers is not in the document]
Comment (ThomasAFink) [reply zmscitizenview/src/styles/breakpoints.scss:87]: @coderabbitai are you sure < >?

File: zmscitizenview/tests/unit/CalendarView.spec.ts [hunk 2]
```diff
@@ -1795,6 +1792,262 @@ describe("CalendarView", () => {
...
+
+      wrapper.vm.selectedProviders = { "2": true };
+      await nextTick();
+
+      expect(wrapper.vm.listViewCurrentHour.has(dateString)).toBe(true);
+      expect(wrapper.vm.listViewCurrentDayPart.has(dateString)).toBe(true);
+      
+      expect(wrapper.vm.listViewCurrentHour.get(dateString)).toBe(16);
+      expect(wrapper.vm.listViewCurrentDayPart.get(dateString)).toBe("pm");
+    });
```
[the remark this answers is not in the document]
Comment (ThomasAFink) [reply zmscitizenview/tests/unit/CalendarView.spec.ts:1874]: @coderabbitai wrong the day remains open when providers are unselected or selected.

File: zmscitizenview/tests/unit/CalendarView.spec.ts [hunk 3]
```diff
@@ -1795,6 +1795,272 @@
...
+        dateString, 
+        hourRows: [{ hour: 15, times: [1], officeId: 1 }, { hour: 16, times: [1], officeId: 1 }] 
+      } as any, "hour");
+      await nextTick();
+
+      // Verify navigation state changed
+      const currentHour = wrapper.vm.listViewCurrentHour.get(dateString);
+      expect(currentHour).toBeDefined();
+      expect(currentHour).toBe(15); // Should have changed to 15
+    });
```
[the remark this answers is not in the document]
Comment (ThomasAFink) [reply zmscitizenview/tests/unit/CalendarView.spec.ts]: @coderabbitai again wrong. A provider could be unselected and then the navigation moves the hours or time of day to the next provider with appointments on that day.
